daemon/cluster/listen_addr.go

```diff
@@ -0,0 +1,163 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func resolveListenAddr(specifiedAddr string, defaultListenAddr string) (string, error) {
+	specifiedHost, specifiedPort, err := net.SplitHostPort(specifiedAddr)
+	if err != nil {
+		return "", fmt.Errorf("could not parse listen address %s", specifiedAddr)
```

Comment:
I'm a bit surprised by this error path: isn't it acceptable to pass an empty listen address and rely on the default?